includes/Core/REST_API/REST_Routes.php

```diff
@@ -93,6 +93,33 @@ function( $server ) {
 				$this->register_routes( $server );
 			}
 		);
+
+		add_filter(
+			'do_parse_request',
+			function( $do_parse_request, $wp ) {
+				add_filter(
+					'query_vars',
+					function( $vars ) use ( $wp ) {
+						// Unsets standard public query vars to escape conflicts between WorPress core
+						// and Google Site Kit APIs which happen when WordPress incorrectly parses request
+						// arguments.
+						$namespace = rest_get_url_prefix() . '/' . self::REST_ROOT;
+						if ( substr( $wp->request, 0, strlen( $namespace ) ) === $namespace ) {
+							// List of variable names to remove from public query variables list.
+							$unset_vars = array( 'orderby' );
+							foreach ( $unset_vars as $unset_var ) {
+								unset( $vars[ array_search( $unset_var, $vars, true ) ] );
+							}
```

Comment:
This seems a but more complex than necessary. How about this?
```suggestion
							return array_values( array_diff( $vars, $unset_vars ) );
```

Reply:
@aaemnnosttv it seems to be easier but it makes more changes than we need. The task is to unset certain variables, your change changes `$vars` array in a whole. It can have a potential negative effect in the future when we come across a plugin that adds its own query vars using stringular keys ( `$query_vars['myplugin_var'] = 'myplugin_var';` ) and relies on those keys somewhere else. We can't be sure which edge cases might be there, so I would encourage not to change the query vars array and leave it as is.

Reply:
@eugene-manuilov I see what you mean, but I don't think we need to be so cautious to protect plugins that are registering query vars the wrong way. `$public_query_vars`, as well as the filtered value of [`query_vars`](https://developer.wordpress.org/reference/hooks/query_vars/) is defined as a list of strings (`@var string[]`) so even though a plugin could do that, we shouldn't go out of our way to preserve it – especially since this is only changed for Site Kit REST requests.

Reply:
Okay, updated.

Reply:
Let's use [`$wp->remove_query_var( $name )`](https://developer.wordpress.org/reference/classes/wp/remove_query_var/) instead 👍

Reply:
@aaemnnosttv we can't use `$wp->remove_query_var( $name )` here because everything that we remove with this function will be overriden with the results of the `query_vars` filter. Do you want me to revert your suggestion and use the initial foreach loop?